### DISCUSSION TITLE
Workaround for lack of IDE Tool menu support for Little Pico Tool?

### DISCUSSION THREAD

**Discussion by dennisma** [earlephilhower/arduino-pico#1732] [Q&A] [answered]
It appears that the IDE 2.* doesn't allow one to add tools to the IDE's Tools menu (see [https://github.com/arduino/arduino-ide/issues/58](https://github.com/arduino/arduino-ide/issues/58) ).   

I didn't realize this until following the install instructions for the Little Pico Tool (see https://github.com/earlephilhower/arduino-pico-littlefs-plugin#installation ).

Are there any good workarounds to this?   I do still have the old IDE and it does show up there so that is one workaround.

Thanks.

**Comment by earlephilhower** [comment]
I think literally 48 hours ago there was a posting/PR merge in the IDE about this.  IDE 2.x is an Arduino wrapper around a wrapper (Theia IDE) around VSCode, and VSCode does allow extensions.  But until a user contributed a method for interacting w/the Arduino board info, options, etc. you couldn't actually do what the old plugins did since it couldn't even, for example, get the current serial port selection.  2.1.1 and above should expose this capability to VSCode extensions.

Unfortunately VSCode runs under Node.js and extensions are written in TypeScript (or possibly JS for purists) and involve about 1,000 dependencies and an "interesting" coding style I'm not sure I quite grok just yet.  And then there's the VSCode specific bits which are (beyond) amply documented by MSFT.

It'll come, but I have a feeling I'm going to need to lean heavily on someone else's extension to start (like I did for the Java based ones here and on the 8266).

Alternatively, if there are VS Code extension writers who can do the framework and just need me to plug in the `mklittlefs` magic, I'd be happy to collaborate!

**Reply by earlephilhower** [reply]
Ooops, it was 2 months ago that the merge happened, but I don't think the tracker issue for plugins got a ping about it until a couple days back.  The first 2.x extension is out, by the same author as the Arduino class extension to IDE 2.x (i.e. I guess he wrote it in parallel with his other PR to add Arduino boards info):  https://github.com/dankeboy36/esp-exception-decoder

**Comment by earlephilhower** [comment, marked as answer]
Took several hours of hacking, is probably the ugliest Typesript you'll ever see, but it can build and upload a filesystems from the IDE:  https://github.com/earlephilhower/arduino-littlefs-upload

**Reply by dennisma** [reply]
Oh wow thanks I just saw this as I came to ask another question.   You didn't have to go to that trouble I just wondered if there was a workaround besides using the old version of the IDE but thanks for doing this and thanks for the great work!
